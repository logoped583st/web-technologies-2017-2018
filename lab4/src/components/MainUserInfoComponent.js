import React from 'react';
import { connect } from "react-redux";


function MainUserInfoComponent(props) {
    return (
        <div className="MainUser">
            <img src={props.image} alt={props.name} />
            <h2>{props.name}</h2>
            <h3>{props.login}</h3>

        </div>
    );
}

const getState = (state) => {
    console.log(state)
    return {
        name: state.reducerUser.name,
        image: state.reducerUser.image,
        login: state.reducerUser.login,
    };
};

export default connect(getState)(MainUserInfoComponent);