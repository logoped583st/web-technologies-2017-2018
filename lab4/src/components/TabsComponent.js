import React from 'react';
import { connect } from "react-redux";
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';

class TabsComponent extends React.Component {
    constructor() {
        super();
        this.state = { tabIndex: 0 };
    }

    render() {
        return (

            <Tabs selectedIndex={this.state.tabIndex} onSelect={tabIndex => this.setState({ tabIndex })}>
                <TabList>
                    <Tab>Основное</Tab>
                    <Tab>Образование</Tab>
                    <Tab>Контакты</Tab>
                </TabList>
                <TabPanel><button >Edit1</button></TabPanel>
                <TabPanel><button >Edit2</button></TabPanel>
                <TabPanel><button >Edit3</button></TabPanel>
            </Tabs>
        );
    }


}



export default connect(null)(TabsComponent);