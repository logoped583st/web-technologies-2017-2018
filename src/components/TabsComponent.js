import React from 'react';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import TextAreaComponent from './TextAreaComponent';

class TabsComponent extends React.Component {

    constructor() {
        super();
        this.state = { tabIndex: 0 };
    }

    render() {
        return (

            <Tabs selectedIndex={this.state.tabIndex} onSelect={tabIndex => this.setState({ tabIndex })}>
                <TabList>
                    <Tab>Основное</Tab>
                    <Tab>Образование</Tab>
                    <Tab>Контакты</Tab>
                </TabList>
                <TabPanel>
                    <TextAreaComponent id="textArea1" number={1} />
                </TabPanel>
                <TabPanel>
                    <TextAreaComponent id="textArea2" number={2} />
                </TabPanel>
                <TabPanel>
                    <TextAreaComponent id="textArea3" number={3} />
                </TabPanel>
            </Tabs>
        );
    }


}



export default (TabsComponent);