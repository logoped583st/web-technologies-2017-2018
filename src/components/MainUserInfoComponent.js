import React from 'react';
import { connect } from "react-redux";
import TabsComponent from './TabsComponent';


function MainUserInfoComponent(props) {
    return (
        <div className="MainUser">
    
            <div className="image_table">
                <img src={props.image} alt={props.name} />
                <TabsComponent className="Table" />
            </div>
            <h2>{props.name}</h2>
            <h3>{props.login}</h3>
           
        </div>
    );
}

const getState = (state) => {
    return {
        name: state.reducerUser.name,
        image: state.reducerUser.avatar_url,
        login: state.reducerUser.login,
    };
};

export default connect(getState)(MainUserInfoComponent);