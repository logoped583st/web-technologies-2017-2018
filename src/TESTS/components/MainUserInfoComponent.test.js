import React from 'react';
import MainUserInfoComponent from '../../components/MainUserInfoComponent';
import TabsComponent from '../../components/TabsComponent'
import { shallow, mount, configure } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('MainUserInfoComponent component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        reducerUser: {
            avatar_url: 'https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4',
            name: 'STAS',
            login: 'logoped583st'
        }
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('MainUserInfoComponent component rendered succesfully', () => {
        const editComponent = shallow(<TabsComponent />);
        expect(mountToJson(editComponent)).toMatchSnapshot();
        const reducerUser = mount(<MainUserInfoComponent store={store} />);
        expect(mountToJson(reducerUser)).toMatchSnapshot();
        editComponent.find('Tabs').simulate('select', 1);
        expect(editComponent.props().selectedIndex).toBe(1);
    });

    it('MainUserInfoComponent props match store', () => {
        const reducerUser = shallow(<MainUserInfoComponent store={store} />);
        expect(reducerUser.prop('name')).toBe('STAS');
        expect(reducerUser.prop('login')).toBe('logoped583st');
      
    })
});