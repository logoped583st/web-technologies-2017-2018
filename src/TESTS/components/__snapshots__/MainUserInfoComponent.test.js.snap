// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainUserInfoComponent component test MainUserInfoComponent component rendered succesfully 1`] = `
<Tabs
  defaultFocus={false}
  defaultIndex={null}
  forceRenderTabPanel={false}
  onSelect={[Function]}
  selectedIndex={0}
>
  <TabList
    className="react-tabs__tab-list"
  >
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Основное
    </Tab>
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Образование
    </Tab>
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Контакты
    </Tab>
  </TabList>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <TextAreaComponent
      id="textArea1"
      number={1}
    />
  </TabPanel>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <TextAreaComponent
      id="textArea2"
      number={2}
    />
  </TabPanel>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <TextAreaComponent
      id="textArea3"
      number={3}
    />
  </TabPanel>
</Tabs>
`;

exports[`MainUserInfoComponent component test MainUserInfoComponent component rendered succesfully 2`] = `
<Connect(MainUserInfoComponent)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MainUserInfoComponent
    dispatch={[Function]}
    image="https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4"
    login="logoped583st"
    name="STAS"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="MainUser"
    >
      <div
        className="image_table"
      >
        <img
          alt="STAS"
          src="https://avatars3.githubusercontent.com/u/30079690?s=400&u=e3443b749fa4d36979e0d631192f3a7bbadc3eeb&v=4"
        />
        <TabsComponent
          className="Table"
          store=""
        >
          <Tabs
            defaultFocus={false}
            defaultIndex={null}
            forceRenderTabPanel={false}
            onSelect={[Function]}
            selectedIndex={0}
          >
            <UncontrolledTabs
              className="react-tabs"
              focus={false}
              forceRenderTabPanel={false}
              onSelect={[Function]}
              selectedIndex={0}
            >
              <div
                className="react-tabs"
                data-tabs={true}
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                <TabList
                  className="react-tabs__tab-list"
                  key=".0"
                >
                  <ul
                    className="react-tabs__tab-list"
                    role="tablist"
                  >
                    <Tab
                      className="react-tabs__tab"
                      disabledClassName="react-tabs__tab--disabled"
                      focus={false}
                      id="react-tabs-0"
                      key=".0"
                      panelId="react-tabs-1"
                      selected={true}
                      selectedClassName="react-tabs__tab--selected"
                      tabRef={[Function]}
                    >
                      <li
                        aria-controls="react-tabs-1"
                        aria-disabled="false"
                        aria-selected="true"
                        className="react-tabs__tab react-tabs__tab--selected"
                        id="react-tabs-0"
                        role="tab"
                        tabIndex="0"
                      >
                        Основное
                      </li>
                    </Tab>
                    <Tab
                      className="react-tabs__tab"
                      disabledClassName="react-tabs__tab--disabled"
                      focus={false}
                      id="react-tabs-2"
                      key=".1"
                      panelId="react-tabs-3"
                      selected={false}
                      selectedClassName="react-tabs__tab--selected"
                      tabRef={[Function]}
                    >
                      <li
                        aria-controls="react-tabs-3"
                        aria-disabled="false"
                        aria-selected="false"
                        className="react-tabs__tab"
                        id="react-tabs-2"
                        role="tab"
                        tabIndex={null}
                      >
                        Образование
                      </li>
                    </Tab>
                    <Tab
                      className="react-tabs__tab"
                      disabledClassName="react-tabs__tab--disabled"
                      focus={false}
                      id="react-tabs-4"
                      key=".2"
                      panelId="react-tabs-5"
                      selected={false}
                      selectedClassName="react-tabs__tab--selected"
                      tabRef={[Function]}
                    >
                      <li
                        aria-controls="react-tabs-5"
                        aria-disabled="false"
                        aria-selected="false"
                        className="react-tabs__tab"
                        id="react-tabs-4"
                        role="tab"
                        tabIndex={null}
                      >
                        Контакты
                      </li>
                    </Tab>
                  </ul>
                </TabList>
                <TabPanel
                  className="react-tabs__tab-panel"
                  forceRender={false}
                  id="react-tabs-1"
                  key=".1"
                  selected={true}
                  selectedClassName="react-tabs__tab-panel--selected"
                  tabId="react-tabs-0"
                >
                  <div
                    aria-labelledby="react-tabs-0"
                    className="react-tabs__tab-panel react-tabs__tab-panel--selected"
                    id="react-tabs-1"
                    role="tabpanel"
                  >
                    <TextAreaComponent
                      id="textArea1"
                      number={1}
                    >
                      <div
                        className="editArea"
                      >
                        <button>
                          Edit 
                          1
                        </button>
                        <textarea
                          className="textArea"
                          type="text"
                          value=""
                        />
                      </div>
                    </TextAreaComponent>
                  </div>
                </TabPanel>
                <TabPanel
                  className="react-tabs__tab-panel"
                  forceRender={false}
                  id="react-tabs-3"
                  key=".2"
                  selected={false}
                  selectedClassName="react-tabs__tab-panel--selected"
                  tabId="react-tabs-2"
                >
                  <div
                    aria-labelledby="react-tabs-2"
                    className="react-tabs__tab-panel"
                    id="react-tabs-3"
                    role="tabpanel"
                  />
                </TabPanel>
                <TabPanel
                  className="react-tabs__tab-panel"
                  forceRender={false}
                  id="react-tabs-5"
                  key=".3"
                  selected={false}
                  selectedClassName="react-tabs__tab-panel--selected"
                  tabId="react-tabs-4"
                >
                  <div
                    aria-labelledby="react-tabs-4"
                    className="react-tabs__tab-panel"
                    id="react-tabs-5"
                    role="tabpanel"
                  />
                </TabPanel>
              </div>
            </UncontrolledTabs>
          </Tabs>
        </TabsComponent>
      </div>
      <h2>
        STAS
      </h2>
      <h3>
        logoped583st
      </h3>
    </div>
  </MainUserInfoComponent>
</Connect(MainUserInfoComponent)>
`;
